openedx/features/edly/utils.py

```diff
@@ -0,0 +1,218 @@
+
+    email_params = get_email_params(xblock)
+    email_params['old_content'] = old_content_with_absolute_urls
+    email_params['new_content'] = new_content_with_absolute_urls
+    students = get_course_enrollments(xblock.location.course_key)
+    send_bulk_mail_to_students.delay(students, email_params, 'handout_changes')
+
+
+def _replace_relative_urls_with_absolute_urls(content, relative_urls, absolute_urls):
+    """
+    This function will replace the all relative url from the given content to the absolute urls
+
+    Arguments:
+        content: Content to be changed
+        relative_urls: List of relative urls to change from content
+        absolute_urls: List of absolute urls to change with relative urls.
+
+    Returns:
+        Updated content contains all absolute urls.
+    """
+    relative_urls = ['"{path}"'.format(path=path) for path in relative_urls]
+    absolute_urls = ['"{path}"'.format(path=path) for path in absolute_urls]
+    for x in range(0, len(relative_urls)):
+        content = content.replace(relative_urls[x], absolute_urls[x])
+    return content
+
+
+def _get_relative_urls(content):
+    """
+    This function will extract the relative urls from content
+
+    Arguments:
+        content: String from which we have to extract the relative imports
+
+    Returns:
+        List of relative urls
+    """
+    pattern = r'href' '*=' '*("/[:.A-z0-9/+@-]*")'
+    try:
+        relative_urls = re.findall(pattern, content)
+        relative_urls = [url.replace('"', '') for url in relative_urls]
+    except TypeError:
+        # If new course created the old_content will be None or Empty
+        return []
+    return relative_urls
+
+
+def _create_absolute_urls(course_key, relative_urls):
+    """
+    This function will make the absolute urls of the given relative urls
+
+    Arguments:
+        course_key: Course key for the assets
+        relative_urls: list of relative urls
+
+    Returns:
+        list of absolute urls
+    """
+    absolute_urls = []
+    for relative_url in relative_urls:
+        if _is_static_path(relative_url):
+            absolute_url = _generate_absolute_url_from_static_path(course_key, relative_url)
+            absolute_urls.append(absolute_url)
+        elif _is_canonicalized_asset_path(relative_url):
+            absolute_url = relative_url.replace('block@', 'block/')
+            absolute_urls.append(absolute_url)
+        else:
+            # This url doesn't belong to the course we are not changing it.
+            relative_urls.remove(relative_url)
+    absolute_urls = ["{lms_base_url}{path}".format(lms_base_url=settings.LMS_ROOT_URL, path=path)
+                     for path in absolute_urls]
+    return relative_urls, absolute_urls
+
+
+def _is_static_path(path):
+    return path.startswith('/static/')
+
+
+def _is_canonicalized_asset_path(path):
+    return path.startswith('/asset')
+
+
+def _generate_absolute_url_from_static_path(course_key, path):
```

Comment:
I think we should try to use any existing method to determine url path of static data. Like using `replace_static_urls` method.